xdsl/utils/lexer.py

```diff
@@ -347,6 +348,18 @@ def get_float_value(self):
         This function will raise an exception if the token is not a float
         literal.
         """
+        if self.kind == Token.Kind.INTEGER_LIT and self.text[:2] in ["0x", "0X"]:
+            match len(self.text):
+                # the number corresponds to `len('0x') + (float_bitwidth / 4)`
+                case 6:
+                    return struct.unpack("<e", struct.pack("<H", int(self.text, 16)))[0]
+                case 10:
+                    return struct.unpack("<f", struct.pack("<I", int(self.text, 16)))[0]
+                case 18:
+                    return struct.unpack("<d", struct.pack("<Q", int(self.text, 16)))[0]
+                case _:
+                    # todo support further bitwidths
+                    pass
```

Comment:
shouldn't the bitwidth be determined by the floating point type? I would expect this to be in a helper function that takes an int and a bitwidth, possibly in the bitwise_casts file, with dedicated tests, and for the logic to be done inline in `parse_optional_number`

Reply:
Good spot, let me fix that

Reply:
Hadn't really considered implied leading zeros, but mlir-opt supports it